react/src/components/timeline-app/TimelineComponent.jsx

```diff
@@ -0,0 +1,177 @@
+import * as ReactDOM from 'react-dom';
+import React from 'react';
+import vis from 'vis';
+import '../../../node_modules/vis/dist/vis.min.css';
+import '../../../scss/timeline-app.scss';
+
+
+export default class TimelineComponent extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+
+  componentDidMount() {
+    if (this.props.data) {
+      this.drawTimeline(this.props.data.events);
+    }
+  }
+
+
+  componentWillReceiveProps(nextProps) {
+    this.drawTimeline(nextProps.data.events);
+  }
+
+
+  /**
+   * @param {Array} events
+   */
+  drawTimeline(events) {
+    /* Prepare the data */
+    let idCounter = 0;
+    /* eslint-disable no-shadow */
+    const { max, min } = events.reduce(({ max, min }, event) => {
```

Comment:
I struggle to understand the purpose of this code snippet, can you abstract it in an own function with meaningful name?